cmd/initContainer/main.go

```diff
@@ -59,7 +59,7 @@ const (
 
 func main() {
 	klog.InitFlags(nil)
-	log.SetLogger(klogr.New())
+	log.SetLogger(klogr.New().WithCallDepth(1))
 	// arguments
```

Comment:
See https://pkg.go.dev/github.com/go-logr/logr#Logger.WithCallDepth

Reply:
I don't know this well enough, it looks like `WithCallStackHelper` could be useful but I don't know exactly how to use it.
Maybe it can be done in another PR ? At least this one fixes the issue we have today.

Reply:
I believe it should be as simple as calling  WithCallStackHelper which returns a new logger and helper. We need to call the helper function and then set the logger.


So

```go
helper, logger := klogr.New().WithCallStackHelper()
helper()
log.SetLogger(logger)
```

Reply:
Ok, I'll try that.

Reply:
I looked again and I don't think `WithCallStackHelper` is adapted to our use case:
> This is useful for users who have helper functions between the "real" call site and the actual calls to Logger methods and want to support loggers which depend on marking each individual helper function, like loggers based on testing.T.

This not our case, we are not wrapping the `Logger`.

Reply:
Happy to approve it. I was able to achieve appropriate results with either of the calls.

My comment was based off -

> To skip one level, WithCallStackHelper() should be used instead of WithCallDepth(1) because it works with implementions that support the CallDepthLogSink and/or CallStackHelperLogSink interfaces.